spec/models/accounting/quickbooks/transaction_creator_spec.rb

```diff
@@ -6,24 +6,126 @@
   let(:class_service) { instance_double(Quickbooks::Service::Class, find_by: [class_ref]) }
   let(:connection) { instance_double(Accounting::Quickbooks::Connection) }
   let(:account) { instance_double(Accounting::Account, qb_id: 98) }
-  subject { described_class.new(instance_double(Division, qb_connection: connection, principal_account: account)) }
+  let(:creator) { described_class.new(instance_double(Division, qb_connection: connection, principal_account: account)) }
 
   before do
-    allow(subject).to receive(:service).and_return(generic_service)
-    allow(subject).to receive(:class_service).and_return(class_service)
+    allow(generic_service).to receive(:create)
+    allow(creator).to receive(:service).and_return(generic_service)
+    allow(creator).to receive(:class_service).and_return(class_service)
   end
 
-  it 'should create class ref, and create journal entry' do
-    expect(generic_service).to receive(:create)
-    expect(class_service).to receive(:find_by)
-
-    subject.add_disbursement(
-      amount: 12.09,
-      loan_id: 2,
-      memo: 'I am memo',
-      description: 'I am a line item description',
-      qb_bank_account_id: 89,
-      qb_customer_id: 3
-    )
+  context 'when qb customer does exist' do
+    let(:qb_customer_id) { 3 }
+    let(:loan_id) { 2 }
+    let(:amount) { 78.20 }
+    let(:memo) { 'I am a memo' }
+    let(:description) { 'I am a line item description' }
+
+    subject do
+      creator.add_disbursement(
+        amount: amount,
+        loan_id: loan_id,
+        memo: memo,
+        description: description,
+        qb_bank_account_id: 89,
+        qb_customer_id: qb_customer_id
+      )
+    end
+
+    context 'generic_service' do
```

Comment:
It's great to beef up specs! I do think these are way too long for what they are testing. Really they are all just checking that the data got assigned properly.

Perhaps a good rule of thumb is that the size of the test code should roughly scale according to the amount of functionality being tested..?

Here is how I would do this:

```
  it 'creates JournalEntry with correct data' do
    expect(generic_service).to receive(:create) do |arg|
      expect(arg.private_note).to eq memo
      expect(arg.txn_date).to be_nil
      lis = arg.line_items
      expect(lis.map { |i| i.journal_entry_line_detail.entity.type }.uniq).to eq %w(Customer)
      expect(lis.map { |i| i.journal_entry_line_detail.entity.entity_ref.value }.uniq).to eq [qb_customer_id]
      expect(lis.map { |i| i.posting_type }).to eq %w(Debit Credit)
      expect(lis.map { |i| i.amount }.uniq).to eq [amount]
      expect(lis.map { |i| i.description }.uniq).to eq [description]
      expect(lis.map { |i| i.qb_account_id }).to eq [XXX, XXX]
      expect(lis.map { |i| i.qb_customer_ref }.uniq).to eq [XXX]
    end
    subject
  end
```

Notice how I've added account_id, posting_type, and customer_ref. Why not!? Account ID is especially important since it varies from one to the other.

Thoughts?

To the extent that you are testing different variations on this, you could move the chunk of expects into a helper method, or just test the things that you expect to vary each time.